Sources/NIOFileSystem/Internal/ParallelRemoval.swift

```diff
@@ -0,0 +1,85 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2024 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIOCore
+
+@available(macOS 10.15, iOS 13.0, watchOS 6.0, tvOS 13.0, *)
+extension FileSystem {
+    /// Iterative implementation of a recursive parallel removal of objects
+    /// found within `path`.
+    func removeConcurrently(
```

Comment:
This could probably live inside of `NIOFileSystem/FileSystem.swift`. I've got no  hard opinions if this should be here or there.